specs/Deployment/DeploymentAPI.md

```diff
@@ -246,12 +264,164 @@ Here's the previous example extended with the `Show UI on error` option:
+will fail installing that WinAppSDK package. But this update to the WinAppSDK packages can be forced
+by using DeploymentRepairOptions object and setting ForceDeployment option before passing it to this
+API. This option when set will shut down the application(s) associated with WinAppSDK packages,
+update the WinAppSDK packages and restart the application(s).
+
+If this option is set when updating Singleton package, then the out of process com server from the
+Singleton package such as push Notifications is explicitly restarted.
+
+When the API is updating Framework package and ForceDeployment option is set, then all dependent
+packages that are NOT currently in use will be immediately re-installed to refer to the updated
+Framework package and all dependent packages that are currently in use will be re-installed, after
+they shut down, to refer to the updated Framework package.
+
+```C#
+    if (DeploymentManager.GetStatus().Status == DeploymentStatus.PackageInstallRequired)
+    {
+        // Create DeploymentRepairOptions object and set ForceDeployment option
+        // to allow for force updating of the WinAppSDK packages even if they are currently in use.
+        var deploymentRepairOptions = new DeploymentRepairOptions();
+        deploymentRepairOptions.ForceDeployment = true;
+
+        // Repair will always attempt to repair WindowsAppRuntime regardless of it's state.
+        // Repair can sometimes take several seconds to deploy the packages.
+        // These should be run on a separate thread so as not to hang your app while the
+        // packages deploy.
+        var RepairTask = Task.Run(() => DeploymentManager.Repair(deploymentRepairOptions));
+        //...do other work while the repair is running...
+        RepairTask.Wait();
+
+        // Check the result.
+        if (RepairTask.Result.Status != DeploymentStatus.Ok)
+        {
+            // The Repair() has failed.
+            // Do error reporting or gather information for submitting a bug.
+            // Gracefully exit the program or carry on without using the WindowsAppRuntime.
+        }
+    }
+```
+
+#### OnError
+
+Additional behavior is available if an error occurs in `DeploymentManager.Repair()`, per the
+following logic:
+
+```
+DeploymentManager::Repair(...)
+{
+    // Only needed supported in packaged processes
+    IF IsPackagedProcess()
+    {
+        // We're not a packaged process. Don't call Repair
+        return OK
+    }
+
+    // Do the initialization work
+    hr = _Repair(...)
+    if FAILED(hr)
+    {
+        LogToEventLog(hr, ...)
+
+        // Should we pop the debugger?
+        IF IsDebuggerPresent()
+        {
+            DebugBreak()
+        }
+
+        // Should we pop some UI?
+        if options.OnError_ShowUI
+        {
+            MessageBox(...)
+        }
+    }
+    return hr
+}
+```
+
+Here's the previous example extended with the `Show UI on error` option:
```

Comment:
This example doesn't match the above logic; the logic shows UI, the example doesn't. If the UI built in to the API and explains to the user what the problem is? Or if the logic is correct and the app needs to show UI, it's awkward because they can't use WinAppSDK to show the UI

Reply:
Yes, the UI shown is built into the API (it's currently a MessageBox())